### PR TITLE
Proxy

### PR DESCRIPTION
**Pull request by ajmaidak**

### CODE REVIEW

File: lib/puppet/provider/gnupg_key/gnupg.rb [hunk 1]
```diff
@@ -53,7 +53,11 @@ def add_key
   end
 
   def add_key_from_key_server
-    command = "gpg --keyserver #{resource[:key_server]} --recv-keys #{resource[:key_id]}"
+    if resource[:proxy].empty?
```
Comment (jwstric2) [inline lib/puppet/provider/gnupg_key/gnupg.rb]: In my environment, I had to set this to .nil?  With .empty? I received the following:

Debug: Executing 'gpg --list-keys --with-colons D39DC0E3'
Error: Could not set 'present' on ensure: undefined method `empty?' for nil:NilClass at 13:/root/sysadmin-puppet/modules/rvm/manifests/gnupg_key.pp
Error: Could not set 'present' on ensure: undefined method `empty?' for nil:NilClass at 13:/root/sysadmin-puppet/modules/rvm/manifests/gnupg_key.pp
Wrapped exception:
undefined method `empty?' for nil:NilClass
Error: /Stage[main]/Rvm::Gnupg_key/Gnupg_key[rvm_D39DC0E3]/ensure: change from absent to present failed: Could not set 'present' on ensure: undefined method `empty?' for nil:NilClass at 13:/root/sysadmin-puppet/modules/rvm/manifests/gnupg_key.pp

After setting to "if resource[:proxy].nil?" receive the below:

Debug: Executing 'gpg --list-keys --with-colons D39DC0E3'
Debug: Executing 'gpg --keyserver hkp://keys.gnupg.net --keyserver-options http-proxy=http://myproxy.company.com:80 --recv-keys D39DC0E3'

ruby 2.0.0p598 (2014-11-13) [x86_64-linux]
puppet 3.8.7

Comment (jwstric2) [inline lib/puppet/provider/gnupg_key/gnupg.rb]: Thanks for the work on this, minus the 1 note pertaining to ?empty ?nil on the resource it works great.

**Comment by phathocker** [comment]
Is this PR ready to merge?  I'd love proxy support

**Comment by edestecd** [comment]
We need proxy support... Ping

**Comment by n1tr0g** [comment]
sure ... i will release the new version this week.

**Comment by edestecd** [comment]
This works well in our environment on CentOS 7.2

**Comment by edestecd** [comment]
I'm getting the error about empty? now as well.  We recently switched to puppet 4, so maybe that's it.

**Comment by edestecd** [comment]
I think empty works in puppet 3 and nil works in puppet 4...
I'm submitting a PR to fix this now....

**Comment by jwstric2** [comment]
@edestecd I hit this in 3.8.7 puppet with ruby 2.0.0 .. see above for error (seems the same from your description) with potential fix (we are running this diff with fix in our production env).  From above comment:



-----------------------------------------------------
In my environment, I had to set this to .nil? With .empty? I received the following:

Debug: Executing 'gpg --list-keys --with-colons D39DC0E3'
Error: Could not set 'present' on ensure: undefined method empty?' for nil:NilClass at 13:/root/sysadmin-puppet/modules/rvm/manifests/gnupg_key.pp Error: Could not set 'present' on ensure: undefined methodempty?' for nil:NilClass at 13:/root/sysadmin-puppet/modules/rvm/manifests/gnupg_key.pp
Wrapped exception:
undefined method empty?' for nil:NilClass Error: /Stage[main]/Rvm::Gnupg_key/Gnupg_key[rvm_D39DC0E3]/ensure: change from absent to present failed: Could not set 'present' on ensure: undefined methodempty?' for nil:NilClass at 13:/root/sysadmin-puppet/modules/rvm/manifests/gnupg_key.pp

After setting to "if resource[:proxy].nil?" receive the below:

Debug: Executing 'gpg --list-keys --with-colons D39DC0E3'
Debug: Executing 'gpg --keyserver hkp://keys.gnupg.net --keyserver-options http-proxy=http://myproxy.company.com:80 --recv-keys D39DC0E3'

ruby 2.0.0p598 (2014-11-13) [x86_64-linux]
puppet 3.8.7
-----------------------------------------------------

**Pull request merged**